sstables_loader.cc

```diff
@@ -582,6 +582,7 @@ class sstables_loader::download_task_impl : public tasks::task_manager::task::im
     sstables_loader::stream_scope _scope;
     std::vector<sstring> _sstables;
     std::vector<shared_ptr<stream_progress>> _progress_per_shard;
```

Comment:
Is it possible to make it `std::vector<foreign_ptr<shared_ptr<stream_progress>>>` instead?

Reply:
I tried. But abandoned the work in progress branch, because of the limit of foreign_ptr. I was not able to come up with an acceptable API interface allowing me to pass a smart pointer down to the delegated API.

Reply:
What about sharded<stream_progress>?

Reply:
> What about sharded<stream_progress>?

good idea! but instead of using `stream_progress`, i am using `shared_ptr<stream_progress>`. let me quote from the comment in the code to explain the rationales:
```c++
        // Wrap stream_progress in a smart pointer to enable polymorphism.
        // This allows derived progress types to be passed down for per-tablet
        // progress tracking while maintaining the base interface.
```